.semaphore/semaphore.yml

```diff
@@ -0,0 +1,42 @@
+version: v1.0
+name: "Semaphore pipeline for alfred-margaret"
+
+agent:
+  machine:
+    type: e1-standard-2
+    os_image: ubuntu1804
+
+# Automatically cancel jobs that are already running for all branches but master
+auto_cancel:
+  running:
+    when: "branch != 'master'"
+
+blocks:
+  - name: Stack
+    task:
+      prologue:
+        commands:
+          # Checkout out the repository
+          - checkout
+          - git status
+          # Restore .stack-work and ~/.stack from cache for faster builds
+          - cache restore home-stack-$SEMAPHORE_GIT_BRANCH
+          - cache restore stack-work-$SEMAPHORE_GIT_BRANCH
+          # Install stack, upgrade it to latest version and add it to PATH
+          - sudo apt-get install haskell-stack
+          - stack upgrade
+          - export PATH=~/.local/bin:$PATH
+
+      jobs:
+        - name: Build alfred-margaret
+          commands:
+            - stack build
```

Comment:
`stack test` already implies a `stack build` (It's just an alias for `stack build --test`), so we don't need this job.

Reply:
I've thought about this too. My thought process was that it would be nice to see at a glance if the build broke or the tests.